src/Controller/TrickController.php

```diff
@@ -14,12 +19,40 @@ class TrickController extends AbstractController
      * View single trick
      *
      * @Route ("/trick/{slug}" , name="trick_detail")
-     * @param string $slug
+     * @param Request $request
+     * @param Trick $trick
      * @return Response
      */
-    public function view(string $slug): Response
+    public function view(Request $request, Trick $trick): Response
     {
-        return $this->render('trick/view.html.twig');
+        if (!$trick) {
+            throw $this->createNotFoundException('Cette figure n\'existe pas');
+        }
+
+        $commentForm = $this->createForm(CommentType::class);
+        $commentForm->handleRequest($request);
+
+        if ($commentForm->isSubmitted() && $commentForm->isValid()) {
+            /** @var Comment $comment */
+            $comment = $commentForm->getData();
+            /** @var User $user */
+            $user = $this->getUser();
```

Comment:
not set a var to user just say directly               $comment->setUser($this->getUser());